masonry/src/tree_arena.rs

```diff
@@ -252,14 +271,47 @@ impl<'a, Item> ArenaRefChildren<'a, Item> {
         self.children
             .iter()
             .find(|child| child.id == id)
-            .map(|child| child.arena_ref(self.id))
+            .map(|child| child.arena_ref(self.id, self.parents_map.parents_map))
     }
 
     // TODO - This method could not be implemented with an actual arena design.
     // It's currently used for some sanity-checking of widget code, but will
     // likely be removed.
     pub(crate) fn iter_children(&self) -> impl Iterator<Item = ArenaRef<'_, Item>> {
-        self.children.iter().map(|child| child.arena_ref(self.id))
+        self.children
+            .iter()
+            .map(|child| child.arena_ref(self.id, self.parents_map.parents_map))
+    }
+
+    /// Find an item in the tree.
+    ///
+    /// Returns a shared reference to the item.
+    ///
+    /// ## Complexity
+    ///
+    /// O(Depth). In future implementations, this will be O(1).
+    pub fn find(self, id: u64) -> Option<ArenaRef<'a, Item>> {
```

Comment:
What's the behaviour here if this is passed `id` of `self`?

Reply:
It returns None. By construction, ArenaRefChildren can't ever give a reference to the current borrow: they're disjoint. I guess I could document that.